### PR TITLE
Add #valid_move? to Rook

### PR DESCRIPTION
**Pull request by saroar**
please who like to check my code i have rubocop problem it was not before whats going on ?

### CODE REVIEW

File: spec/models/rook_spec.rb [hunk 1]
```diff
@@ -4,4 +4,7 @@
   it 'has type of Rook' do
     expect(subject.type).to eq 'Rook'
   end
+
+  describe '#valid_move?' do
+  end
```
Comment (jonmast) [inline spec/models/rook_spec.rb]: You should add some specs here. Check queen_spec or bishop_spec for ideas.

**Comment by jonmast** [comment]
:+1: Looks good.

File: app/models/rook.rb [hunk 2]
```diff
@@ -1,2 +1,8 @@
 class Rook < ChessPiece
+  def valid_move?(destination_x, destination_y)
+    return false if out_of_bounds?(destination_x, destination_y)
+    return false if diagonal_move?(destination_x, destination_y)
+    return false if obstructed?(destination_x, destination_y)
+    horizontal_move?(destination_x, destination_y) || vertical_move?(destination_x, destination_y) ? true : false
```
Comment (jonmast) [inline app/models/rook.rb]: You should never have to use a ternary to return true and false. In this case you can just remove the ternary and get the exact same result.

File: spec/models/rook_spec.rb [hunk 3]
```diff
@@ -4,4 +4,34 @@
...
+    it 'returns true for horizontal moves' do
+      expect(rook.valid_move?(4, 5)).to eq true
+    end
+
+    it 'returns false for diagonal moves' do
+      expect(rook.valid_move?(4, 4)).to eq false
+    end
+
+    it 'returns false for jagged moves' do
+      expect(rook.valid_move?(6, 4)).to eq false
```
Comment (jonmast) [inline spec/models/rook_spec.rb]: This is actually a diagonal move.

File: app/models/rook.rb [hunk 4]
```diff
@@ -1,2 +1,8 @@
 class Rook < ChessPiece
+  def valid_move?(destination_x, destination_y)
+    return false if out_of_bounds?(destination_x, destination_y)
+    return false if diagonal_move?(destination_x, destination_y)
+    return false if obstructed?(destination_x, destination_y)
```
Comment (jonmast) [inline app/models/rook.rb]: I think obstructed? needs to be at the end.

**Comment by radiofreemattd** [comment]
@jonmast agree with all of the above. Should be more readable now, as well as consistent with the other piece models.

**Comment by mscoutermarsh** [comment]
nice!

**Pull request merged**